### PR TITLE
Pmannan/nemo ft update

### PR DESCRIPTION
**Pull request by parthmannan**
# What does this PR do ?

Adding finetuning script and config.

### CODE REVIEW

File: nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py [hunk 1]
```diff
@@ -467,8 +471,10 @@ def forward(
         media = kwargs.pop('media', None)
         if parallel_state.is_pipeline_first_stage(ignore_virtual=True):
             self.embedding.word_embeddings.set_media(media)
+        torch.cuda.nvtx.range_push("LLM Model")
         result = GPTModel.forward(self, *args, **kwargs)
         torch.cuda.nvtx.range_pop()
+        torch.cuda.nvtx.range_pop()
```
Comment (tfogal) [inline nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py:477]: this is popped twice, now

File: neva.sh [hunk 2]
```diff
@@ -56,4 +56,4 @@ python3 \
 rm -fr ${TMPDIR}
 if test -f ${profile} ; then
   cp ${profile} ~/share/
-fi
+fi
```
Comment (tfogal) [inline neva.sh:59]: is your editor maybe dropping EOL whitespace? i'm not sure how to read github's ⛔ icon?
anyway not a big deal

File: dynamo-neva-finetune.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+# /scratch/nsight-systems-linux-public-DVS/target-linux-x64/nsys profile -w true -t cublas,cuda,nvtx,osrt -s cpu -c cudaProfilerApi -o dynamo_thunder_neva_finetune \
+
+HYDRA_FULL_ERROR=1 \
+THUNDER_ANNOTATE_TRACES=1 \
+NEMO_THUNDER_NEVA=dynamo \
+python3 \
+	./examples/multimodal/multimodal_llm/neva/neva_finetune.py \
+         --config-path=/scratch/thunder_neva/tfogal_nemo_neva \
```
Comment (tfogal) [inline dynamo-neva-finetune.sh:33]: i'll see if we can use `./` for the path so that we don't have to force people to have hardcoded paths

**Review by tfogal — APPROVED** [review]
this is great, thank you!

i'm tempted to just merge as-is, but a couple questions:
- does the config to use llama3_8b mean it no longer fits on a single GPU? for people also use this branch for just non-perf debugging thunder+nemo on their workstations.
- there is a duplicate range_pop, unless I missed something?

File: nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py (hunk 1 above)
Comment (parthmannan) [reply nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py:477]: One pop is for the `LLM Model` and one pop is for the `neva_fwd`

File: dynamo-neva-finetune.sh (hunk 3 above)
Comment (parthmannan) [reply dynamo-neva-finetune.sh:33]: Ah yes. I'll check this as well if I can grab a node.

File: nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py (hunk 1 above)
Comment (tfogal) [reply nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py:477]: ooh, oops, the diff rendered weird and it broke my brain. sorry about that!

**Comment by parthmannan** [comment]
>does the config to use llama3_8b mean it no longer fits on a single GPU? for people also use this branch for just non-perf debugging thunder+nemo on their workstations.

The num_layers has been adjusted to run on a single GPU. There is definitely room to make a better config such as reducing micro_batch_size and increasing num_layers for a more realistic example but the config should run on single GPU right now.

**Pull request merged**